### PR TITLE
Bug fixing on Python 3.8

### PR DESCRIPTION
**Pull request by rassouly**
A couple random bug fixes and a migration from Unicode to Str

### CODE REVIEW

**Comment by rassouly** [comment]
The errors on the Python 3.6 version are exactly the ones I fixed by migrating all the `Unicode `fields to `Str`. Since the errors are only in the conversion code which is completely unused by now, it might be ok to merge anyway.

EDIT: I think the errors come from exopy_pulses which I still have not migrated to `Str`

**Comment by MatthieuDartiailh** [comment]
I agree that the conversion code has no real value anymore. I would be fine with it being removed from the repo (perhaps in its own PR). It would be worth bumping the tested version of Python to 3.6, 3.7 and 3.8 also.

File: exopy_hqc_legacy/tasks/tasks/instr/dc_tasks.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class SetDCVoltageTask(InterfaceableTaskMixin, InstrumentTask):
 
     """
     #: Target value for the source (dynamically evaluated)
-    target_value = Unicode().tag(pref=True,
+    target_value = Str().tag(pref=True,
                                  feval=validators.SkipLoop(types=numbers.Real))
```
Comment (MatthieuDartiailh) [inline exopy_hqc_legacy/tasks/tasks/instr/dc_tasks.py]: improper indent

File: exopy_hqc_legacy/tasks/tasks/instr/dc_tasks.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ class SetDCCurrentTask(InterfaceableTaskMixin, InstrumentTask):
...
```
Comment (MatthieuDartiailh) [inline exopy_hqc_legacy/tasks/tasks/instr/dc_tasks.py]: Same as above

File: exopy_hqc_legacy/tasks/tasks/instr/oscilloscope_tasks.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ class OscilloGetTraceTask(InstrumentTask):
     trace = Enum('1', '2', '3', '4', 'TA', 'TB', 'TC', 'TD').tag(pref=True)
 
     #: Number of time the instrument should average.
-    average_nb = Unicode().tag(pref=True,
+    average_nb = Str().tag(pref=True,
                                feval=validators.Feval(types=numbers.Integral))
```
Comment (MatthieuDartiailh) [inline exopy_hqc_legacy/tasks/tasks/instr/oscilloscope_tasks.py]: Same as above

**Review by MatthieuDartiailh — APPROVED** [review]
Looks good except a bunch of comments regarding indentation.

**Comment by MatthieuDartiailh** [comment]
I think we can drop 3.5 from the test matrix (I believe I already did so for exopy itself). Also when you corrected the indentation, did you check the line length (<89) ?

**Comment by rassouly** [comment]
I've bumped the minimum version to 3.6 and I've checked the line length

**Comment by MatthieuDartiailh** [comment]
Could you skip the tests in conversion to avoid the apparent bad CIs ? Once this is done I will merge. Would you have time for a PR removing the conversion code ?

**Comment by rassouly** [comment]
I've skipped the failing tests but Travis doesn't seem to be able to install exopy on Python 3.7 because apparently exopy_pulses requires a lower version of python.

**Comment by rassouly** [comment]
I'm also preparing a PR to remove all the conversion code. I'll send it once the CI changes are merged

**Comment by MatthieuDartiailh** [comment]
I will try to fix the builds later this week... I forgot how it worked and made a bigger mess... Should be done Wednesday night.

**Comment by rassouly** [comment]
Thank you, have a nice week

**Comment by MatthieuDartiailh** [comment]
Ok the CIs of every repos are now fixed so I am going to merge this now.

**Pull request merged**